packages/opal-common/opal_common/sources/git_policy_source.py

```diff
@@ -49,7 +51,16 @@ def __init__(
             ssh_key=self._ssh_key,
             clone_timeout=request_timeout,
         )
+
+        if branch_name is None and tag_name is None:
+            logger.exception("Must provide either branch_name or tag_name")
```

Comment:
Since we are not within in a `except:` clause it's more correct to use `logger.error` 
See: https://docs.python.org/3/library/logging.html#logging.Logger.exception

Also maybe best to include the actual ENV-VAR name- as the average user will struggle with translating

Reply:
Alternatively you can raise the exception here, and just log it in a higher except clause- which would be nicer,

Reply:
I see, thanks! Would that mean wrapping [this](https://github.com/philipclaesson/opal/blob/279415462e6830480320b85b54e2af379ac3bcfa/packages/opal-server/opal_server/policy/watcher/factory.py#L103-L111) in a try/except?

Reply:
That sounds like a good option, yes